### DISCUSSION TITLE
Let's make a Node.js API for this?

### DISCUSSION THREAD

**Discussion by Anirudh-techie** [charmbracelet/gum#38] [Ideas]
i just saw this repo and i love it. I just really like the idea and i think this would be very useful to be used as a library in a platform like Node.Js

I would love to help btw. anyways byee

**Comment by maaslalani** [comment]
Hey! I put together a little example of how to use `gum` in Node.js.

https://github.com/charmbracelet/gum/blob/main/examples/gum.js

**Reply by maaslalani** [reply]
I think improving the ergonomics of this in Node.js would be quite cool!

```javascript
const choice = gum.choose("Foo", "Bar", "Baz");
```

**Reply by Anirudh-techie** [reply]
Yea, don't you think having a dedicated npm package for this would be pretty cool and personally as a node.js developer i would definitely use it for my projects.

**Reply by maaslalani** [reply]
Yeah I totally agree, we just wouldn't have the time to maintain it and give it the necessary updates. If you would want to make a dedicated npm package. You have our full support!

**Reply by Anirudh-techie** [reply]
Great! I'll try my best to take initiative

**Reply by AdamGaskins** [reply]
@maaslalani I'm thinking about doing a similar thing but for PHP-- how would you recommend we distribute the binaries? Is there an easier way than just downloading all of them, committing them in the git repository, and then calling the correct one depending on the platform?

**Comment by beltagyy** [comment]
There is no official Node.js API for Gum, and the maintainers intend it to stay a standalone Go binary. For now, the best way to use Gum from Node.js is via child_process or execa. If you need a programmatic API, enquirer or @inquirer/prompts are solid native Node alternatives.
